polaris/benchmark/_base.py

```diff
@@ -432,72 +470,32 @@ def evaluate(
-
-        if not isinstance(y_pred, dict) or all(k in self.target_cols for k in y_pred):
-            y_pred = {"test": y_pred}
-
-        if not isinstance(y_prob, dict) or all(k in self.target_cols for k in y_prob):
-            y_prob = {"test": y_prob}
-
-        if any(k not in y_pred for k in test.keys()) and any(k not in y_prob for k in test.keys()):
-            raise KeyError(
-                f"Missing keys for at least one of the test sets. Expecting: {sorted(test.keys())}"
-            )
-
-        # Results are saved in a tabular format. For more info, see the BenchmarkResults docs.
-        scores: ResultsType = pd.DataFrame(columns=BenchmarkResults.RESULTS_COLUMNS)
-
-        # For every test set...
-        for test_label, y_true_subset in y_true.items():
-            # For every metric...
-            for metric in self.metrics:
-                if metric.is_multitask:
-                    # Multi-task but with a metric across targets
-                    score = metric(
-                        y_true=y_true_subset, y_pred=y_pred.get(test_label), y_prob=y_prob.get(test_label)
-                    )
-                    scores.loc[len(scores)] = (test_label, "aggregated", metric, score)
-                    continue
-
-                if not isinstance(y_true_subset, dict):
-                    # Single task
-                    score = metric(
-                        y_true=y_true_subset, y_pred=y_pred.get(test_label), y_prob=y_prob.get(test_label)
-                    )
-                    scores.loc[len(scores)] = (
-                        test_label,
-                        self.target_cols[0],
-                        metric,
-                        score,
-                    )
-                    continue
+        # The `evaluate_benchmark` function expects the benchmark labels to be of a certain structure which
+        # depends on the number of tasks and test sets defined for the benchmark. Below, we build the structure
+        # of the benchmark labels based on the aforementioned factors.
+        test = self._get_test_set(hide_targets=False)
+        if isinstance(test, dict):
+            #
+            # For multi-set benchmarks
+            y_true = {}
+            for test_set_name, values in test.items():
+                y_true[test_set_name] = {}
+                if isinstance(values.targets, dict):
+                    #
+                    # For multi-task, multi-set benchmarks
+                    for task_name, values in values.targets.items():
+                        y_true[test_set_name][task_name] = values
+                else:
+                    #
+                    # For single task, multi-set benchmarks
+                    for task_name in self.target_cols:
+                        y_true[test_set_name][task_name] = values.targets
+        else:
+            #
+            # For single set benchmarks (single and multiple task)
+            y_true = test.targets
```

Comment:
Yes, quite confusing. The idea here is that the structure of the `y_true` object must match that of the `y_pred` and `y_prob` objects, and those depend on the number of tasks and test sets. I hope that comment you found showing examples of this will help users in the time being. 

And I agree, I think the `BenchmarkPredictions` class mentioned in #169 with the validators would be awesome for UX.